airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -400,4 +355,153 @@ private void continueAsNew(final ConnectionUpdaterInput connectionUpdaterInput)
+
+    final ScheduleRetrieverOutput scheduleRetrieverOutput = runMandatoryActivityWithOutput(configFetchActivity::getTimeToWait,
+        scheduleRetrieverInput);
+
+    return scheduleRetrieverOutput.getTimeToWait();
+  }
+
+  /**
+   * Creates a new job if it is not present in the input. If the jobId is specified in the input of
+   * the connectionManagerWorkflow, we will return it. Otherwise we will create a job and return its
+   * id.
+   */
+  private Long getOrCreateJobId(final ConnectionUpdaterInput connectionUpdaterInput) {
+    if (connectionUpdaterInput.getJobId() != null) {
+      return connectionUpdaterInput.getJobId();
+    }
+
+    final JobCreationOutput jobCreationOutput =
+        runMandatoryActivityWithOutput(
+            jobCreationAndStatusUpdateActivity::createNewJob,
+            new JobCreationInput(
+                connectionUpdaterInput.getConnectionId(), workflowState.isResetConnection()));
+    connectionUpdaterInput.setJobId(jobCreationOutput.getJobId());
+
+    return jobCreationOutput.getJobId();
+  }
+
+  /**
+   * Create a new attempt for a given jobId
+   */
+  private Integer createAttemptId(long jobId) {
+    final AttemptCreationOutput attemptCreationOutput =
+        runMandatoryActivityWithOutput(
+            jobCreationAndStatusUpdateActivity::createNewAttempt,
+            new AttemptCreationInput(
+                jobId));
+
+    return attemptCreationOutput.getAttemptId();
+  }
+
+  /**
+   * Generate the input that is needed by the job. It will generate the configuration needed by the
+   * job and will generate a different output if the job is a sync or a reset.
+   */
+  private GeneratedJobInput getJobInput() {
+    final SyncInput getSyncInputActivitySyncInput = new SyncInput(
+        attemptId,
+        jobId,
+        workflowState.isResetConnection());
+
+    final GeneratedJobInput syncWorkflowInputs = runMandatoryActivityWithOutput(
+        getSyncInputActivity::getSyncWorkflowInput,
+        getSyncInputActivitySyncInput);
+
+    return syncWorkflowInputs;
+  }
+
+  /**
+   * Report the job as started in the job tracker and set it as running in the workflow internal
+   * state.
+   */
+  private void reportJobStarting() {
+    runMandatoryActivity(
+        jobCreationAndStatusUpdateActivity::reportJobStart,
+        new ReportJobStartInput(
+            jobId));
+
+    workflowState.setRunning(true);
+  }
+
+  /**
+   * Start the child SyncWorkflow. We are using the sync workflow here for two main reason: -
+   * Originally the Sync workflow was living by himself and was launch by the scheduler. In order to
+   * limit the portential migration issues, we kept the {@link SyncWorkflow} as is and launch it as a
+   * sync workflow. - The {@link SyncWorkflow} has different requirements than the
+   * {@link ConnectionManagerWorkflow} because it mainly runs long term activity, in the future, using
+   * a different Node pool would make sense.
+   */
```

Comment:
```suggestion
   * Start the child SyncWorkflow. We are using a child workflow here for two main reason: 
   * <p>
   * - Originally the Sync workflow was living by himself and was launch by the scheduler. In order to
   * limit the potential migration issues, we kept the {@link SyncWorkflow} as is and launch it as a
   * child workflow. 
   * <p>
   * - The {@link SyncWorkflow} has different requirements than the
   * {@link ConnectionManagerWorkflow} since the latter is a long running workflow, in the future, using
   * a different Node pool would make sense.
   */
```